trieste/space.py

```diff
@@ -387,9 +420,9 @@ def dimension(self) -> TensorType:
         """The number of inputs in this search space."""
         return self._dimension
 
-    def __contains__(self, value: TensorType) -> bool | TensorType:
-        tf.debugging.assert_shapes([(value, self.points.shape[1:])])
-        return tf.reduce_any(tf.reduce_all(value == self._points, axis=1))
+    def _contains(self, value: TensorType) -> TensorType:
+        comparison = tf.math.equal(self._points, tf.expand_dims(value, -2))  # [..., N, D]
```

Comment:
why is value a different shape to self._points?

Reply:
The idea is that `value` can be a single point, an arbitrary list of points, an arbitrary array of points, etc, while `self._points` is the fixed list of points defining the space. The computation needs to check for every point in `value` whether it is equal to one of the points in `self._points`.